### PR TITLE
feat(environments): Add group events environment filtering

### PR DESCRIPTION
**Pull request by lynnagara**
Add ability to filter group events by environment

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/groupEvents.jsx [hunk 1]
```diff
@@ -10,42 +11,90 @@ import Pagination from '../components/pagination';
...
+    if (nextProps.environment !== this.props.environment) {
+      const newQueryString = getQueryStringWithEnvironment(
+        nextProps.location.query.query || '',
+        nextProps.environment ? nextProps.environment.name : null
+      );
+      this.setState(
+        {
+          query: newQueryString,
+        },
+        this.handleSearch(newQueryString)
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/groupEvents.jsx]: Why `setState` here if `handleSearch` is going to navigate to a diff route. Also why call `handleSearch` in callback if it doesn't depend on `state`?

File: src/sentry/static/sentry/app/views/stream.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ import StreamFilters from './stream/filters';
 import StreamSidebar from './stream/sidebar';
 import TimeSince from '../components/timeSince';
 import utils from '../utils';
-import streamUtils from './stream/utils';
+import qsUtils from '../utils/queryString';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/stream.jsx]: nit: let's try to keep import name the same as module name

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/searchBar.jsx [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
     onSearch: PropTypes.func,
-    onQueryChange: PropTypes.func,
     placeholder: PropTypes.string,
   };
 
   static defaultProps = {
     defaultQuery: '',
     query: '',
     onSearch: function() {},
-    onQueryChange: function() {},
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/searchBar.jsx]: What about the component passing `onQueryChange` to this? Was it not being passed?

File: src/sentry/static/sentry/app/views/groupEvents.jsx [hunk 4]
```diff
@@ -10,42 +11,90 @@ import Pagination from '../components/pagination';
...
     this.fetchData();
   },
 
   componentWillReceiveProps(nextProps) {
-    if (
-      nextProps.params.groupId !== this.props.params.groupId ||
-      nextProps.location.search !== this.props.location.search
-    ) {
-      let queryParams = nextProps.location.query;
+    // If query has changed, update the environment with the query environment
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/groupEvents.jsx]: Is it intentional that this is no longer handling the case where `groupId` is changed?

**Review by benvinegar — CHANGES_REQUESTED** [review]
See comments, specifically around handling `groupId` changes. It feels like this should include at least snapshot tests; these are sensitive pages, and introducing changes feels like we should add coverage.

File: src/sentry/static/sentry/app/views/groupEvents.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/groupEvents.jsx]: Good catch, that's not necessary

File: src/sentry/static/sentry/app/components/searchBar.jsx (hunk 3 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/searchBar.jsx]: It wasn't passed or used anywhere

File: src/sentry/static/sentry/app/views/groupEvents.jsx (hunk 4 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/groupEvents.jsx]: Yes, it's guarding against a situation that I can't see ever arising. We don't change group ID within the group details view. To change group ID you'd have to navigate to a different route and the component will be remounted.

Comment (benvinegar) [reply src/sentry/static/sentry/app/views/groupEvents.jsx]: Seems I added it when upgrading to react-router 1.0 [here](https://github.com/getsentry/sentry/commit/b5571593ecb066567dcef3eaf42fac520626c7eb#diff-323aa0c7a08b4dc4c847b8897a152085R60). I don't really remember what I was doing; maybe I wasn't using routes correctly. I agree it should be re-mounted.

**Review by benvinegar — APPROVED** [review]

**Pull request merged**